File: Number of Enclaves/no. of enclaves.java
class Solution {
    public int numEnclaves(int[][] grid) {
        int m = grid.length;
        int n = grid[0].length;
        
        for(int i = 0; i < m; i++) {
            dfsCore(i, 0, grid);
            dfsCore(i, n - 1, grid);
        }
        
        for(int i = 0; i < n; i++) {
            dfsCore(0, i, grid);
            dfsCore(m - 1, i, grid);
        }
        
        int count = 0;
        for(int i = 0; i < m; i++) {
            for(int j = 0; j < n; j++) {
                if (grid[i][j] == 1) {
                    count++;
                }
            }
        }
        
        return count;
    }
    
    public void dfsCore(int row, int col, int[][] grid) {
        if (row >= 0 && row < grid.length && col >=0 && col < grid[0].length &&
           grid[row][col] == 1) {
            grid[row][col] = 0;
            dfsCore(row - 1, col, grid);
            dfsCore(row + 1, col, grid);
            dfsCore(row, col - 1, grid);
            dfsCore(row, col + 1, grid);
        }
    }
}

File: Number of Enclaves/no. of enclaves.py
class Solution:
    def numEnclaves(self, grid: List[List[int]]) -> int:
        m = len(grid)
        n = len(grid[0])
        
        for i in range(m):
            self.dfsCore(i, 0, grid)
            self.dfsCore(i, n - 1, grid)
        
        for i in range(n):
            self.dfsCore(0, i, grid)
            self.dfsCore(m - 1, i, grid)
        
        count = 0
        for i in range(m):
            for j in range(n):
                if (grid[i][j] == 1):
                    count += 1
        
        return count
    
    def dfsCore(self, row, col, grid):
        if (row >= 0 and row < len(grid) and col >= 0 and col < len(grid[0]) and grid[row][col] == 1):
            grid[row][col] = 0
            self.dfsCore(row - 1, col, grid)
            self.dfsCore(row + 1, col, grid)
            self.dfsCore(row, col - 1, grid)
            self.dfsCore(row, col + 1, grid)
    